plugins/CoreHome/vue/src/ReportingMenu/ReportingMenu.vue

```diff
@@ -180,7 +195,16 @@ export default defineComponent({
       return document.querySelector('nav .activateLeftMenu');
     },
     menu() {
-      return ReportingMenuStoreInstance.menu.value;
+      const categories = ReportingMenuStoreInstance.menu.value;
+
+      categories.forEach((category) => {
+        if (category.widget && category.widget.indexOf('.') > 0) {
+          const [widgetPlugin, widgetComponent] = category.widget.split('.');
```

Comment:
```suggestion
        const widgetParts = category.widget.split('.');
        if (category.widget && widgetParts.length ===  2) {
          const [widgetPlugin, widgetComponent] = widgetParts;
```

Just to prevent bugs/issues later when there are `category.widget` values of things like `a.b.c`

Reply:
I can add a limit to the split function so it always produces two parts from the first two, or we use splat to catch all the remaining parts into the component.. or I can add a check to the `setWidget` call that would throw on invalid format inputs... basically I don't think it should be dealt with here in this way, but happy to discuss. It's basically a matter "if someone provides incompatible format widget notation, it won't work for them" and them is mostly us.

Reply:
Maybe in the backend is better `setWidget` can be a run time error at least and stop any headaches going forward. A bit of defensive programming.
Otherwise it could be a value object with two parts `plugin` and `component` and a __toString() method that concats them in the correct format.